ts/Extensions/RangeSelector.ts

```diff
@@ -951,6 +951,8 @@ class RangeSelector {
                 }
             );
         }
+
+        fireEvent(chart, 'afterRangeSelectorBtnClick');
```

Comment:
Is there a good reason for firing the event on the chart instead of the `RangeSelector`? Seems like a bit of a pollution of the `Chart` with something that it shouldn't be concerned with...

Reply:
I initially had it on RangeSelector, but that class is not set up to deal with events when `destroy`ing and such, so it would be a more elaborate refactoring. Should we change it?

Reply:
Yes I think so... I think it would be cleaner

Reply:
PS: If you add it to the RangeSelector class rather than the instance, you should not have to worry about destroying.

```js
addEvent(RangeSelector, 'afterBtnClick', e => {});
```